core/src/main/scala/kafka/zk/ZkMigrationClient.scala

```diff
@@ -47,24 +46,44 @@ import scala.jdk.CollectionConverters._
  */
 class ZkMigrationClient(zkClient: KafkaZkClient) extends MigrationClient with Logging {
 
-  override def getOrCreateMigrationRecoveryState(initialState: ZkMigrationLeadershipState): ZkMigrationLeadershipState = {
-    zkClient.createTopLevelPaths()
-    zkClient.getOrCreateMigrationState(initialState)
+  @throws(classOf[MigrationClientException])
+  def wrapZkException[T](fn: => T): T = {
```

Comment:
Should we make it so that `wrapZkException(wrapZkException(e))` is the same as `wrapZkException(e)`? In case we inadvertently end up nesting these somehow.

(Concretely I'm just saying: let's add a pass-through case for MigrationClientAuthException / MigrationClientException)